queasars/minimum_eigensolvers/base/termination_criteria.py

```diff
@@ -109,3 +110,184 @@ def check_termination(
             return False
 
         return True
+
+
+class BestIndividualExpectationValueThreshold(EvolvingAnsatzMinimumEigensolverBaseTerminationCriterion):
+    """
+    Termination criterion which terminates, if the expectation value of the best individual of a population
+    falls below a certain threshold value
+
+    :param expectation_threshold: expectation value below which the optimization terminates
+    :type expectation_threshold: float
+    """
+
+    def __init__(self, expectation_threshold: float):
+        self._expectation_threshold: float = expectation_threshold
+
+    def reset_state(self) -> None:
+        pass
+
+    def check_termination(
+        self,
+        population_evaluation: BasePopulationEvaluationResult,
+        best_individual: BaseIndividual,
+        best_expectation_value: float,
+    ) -> bool:
+        if population_evaluation.best_expectation_value < self._expectation_threshold:
+            return True
+        return False
+
+
+def _median_hausdorff_distance_by_expectation_value(
+    result_1: BasePopulationEvaluationResult, result_2: BasePopulationEvaluationResult
+) -> float:
+    """
+    Calculates the median Hausdorff distance between the expectation values in two population evaluation results
+
+    :arg result_1: first evaluation result to calculate the distance on
+    :type result_1: BasePopulationEvaluationResult
+    :arg result_2: second evaluation result to calculate the distance on
+    :type result_2: BasePopulationEvaluationResult
+    :return: the median Hausdorff distance between the expectation values in two population evaluation results
+    """
+
+    def distance(from_expectations: list[float], to_expectations: list[float]) -> float:
```

Comment:
These lists are of population size, right? So not too large I guess.

Reply:
Exactly. 👍